pycqed/analysis_v3/data_processing.py

```diff
@@ -577,318 +440,136 @@ def rotate_iq(data_dict, data_keys_in, data_keys_out, **params):
-                    'xvals': cal_swpts[cal_pts_idxs],
-                    'yvals': cal_data[cal_pts_idxs],
-                    'setlabel': list(qb_cal_indxs)[ii],
-                    'do_legend': do_legend_cal_states,
-                    'legend_bbox_to_anchor': (1, 0.5),
-                    'legend_pos': 'center left',
-                    'linestyle': 'none',
-                    'line_kws': {'color': get_cal_state_color(
-                        list(qb_cal_indxs)[ii])}}
-
-                plot_dicts[plot_dict_name_cal+'_line'] = {
-                    'fig_id': fig_name,
-                    'plotsize': plotsize,
-                    'plotfn': 'plot_hlines',
-                    'y': np.mean(cal_data[cal_pts_idxs]),
-                    'xmin': physical_swpts[0],
-                    'xmax': cal_swpts[-1],
-                    'colors': 'gray'}
-
-        title = (data_dict['timestamp'] + ' ' + data_dict['measurementstring'])
-        if title_suffix is not None:
-            title += '\n' + title_suffix
-
-        plot_dict_name = fig_name + '_' + plot_name_suffix
-        # get x info (try for old sequences which do not have info in metadata
-        hard_sweep_params = get_param('hard_sweep_params', data_dict)
-        sweep_name = get_param('sweep_name', data_dict)
-        sweep_unit = get_param('sweep_unit', data_dict)
-        if hard_sweep_params is not None:
-            xlabel = list(hard_sweep_params)[0]
-            xunit = list(hard_sweep_params.values())[0][
-                'unit']
-        elif (sweep_name is not None) and (sweep_unit is not None):
-            xlabel = sweep_name
-            xunit = sweep_unit
-        else:
-            xlabel = data_dict['sweep_parameter_names']
-            xunit = data_dict['sweep_parameter_units']
-        if np.ndim(xunit) > 0:
-            xunit = xunit[0]
-
-        if data_axis_label is None:
-            data_axis_label = '{} state population'.format(
-                get_latex_prob_label(keyi))
-
-        plot_dicts[plot_dict_name] = {
-            'plotfn': 'plot_line',
-            'fig_id': fig_name,
-            'plotsize': plotsize,
-            'xvals': physical_swpts,
-            'xlabel': xlabel,
-            'xunit': xunit,
-            'yvals': data,
-            'ylabel': data_axis_label,
-            'yunit': '',
-            'setlabel': data_label,
-            'title': title,
-            'linestyle': 'none',
-            'do_legend': do_legend_data,
-            'legend_bbox_to_anchor': (1, 0.5),
-            'legend_pos': 'center left'}
-
-    if len(plot_names_cal) > 0:
-        if do_legend_data and not do_legend_cal_states:
-            for plot_name in plot_names_cal:
-                plot_dict_cal = plot_dicts.pop(plot_name)
-                plot_dicts[plot_name] = plot_dict_cal
-
-    if 'plot_dicts' in data_dict:
-        data_dict['plot_dicts'].update(plot_dicts)
-    else:
-        data_dict['plot_dicts'] = plot_dicts
-
```

Comment:
Other general comment on analysis V3: Don't we want a base class for Node ? or would this be meaningless?

Reply:
Since the nodes can be either functions of classes, I don't know if this makes sense. Then you would be enforcing that they are all classes, but most should be functions in the original concept in order to keep things simple. The only things the nodes should have in common are a few input arguments.